modules/tushentsova_k_permut_generator/test/test_permut_generator.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2024 Tushentsova Karina
+
+#include <gtest/gtest.h>
+#include "include/permut_generator.h"
+
+TEST(Tushentsova_Karina_Permut_Generator, CanGetArrCertainLength) {
+    int a[] = { 1, 2, 3 };
+    ASSERT_NO_THROW(PermutGenerator arr(a, 3));
+}
+
+TEST(Tushentsova_Karina_Permut_Generator, CanGetArrInvalidValue) {
+    int a[] = { 1, 2, 3 };
+    ASSERT_THROW(PermutGenerator arr(a, 0), std::string);
+    ASSERT_THROW(PermutGenerator arr(a, -3), std::string);
+    int* p = NULL;
+    ASSERT_THROW(PermutGenerator arr(p, 3), std::string);
+}
+
+TEST(Tushentsova_Karina_Permut_Generator, CanPrintArray) {
+    int const size = 3;
+    int a1[] = { 1, 1, 1 }, a2[] = { 1, 2, 3 };
+
+    PermutGenerator arr1(a1, size);
+    PermutGenerator arr2(a2, size);
+
+    EXPECT_EQ("111", arr1.PrintArray());
+    EXPECT_EQ("123", arr2.PrintArray());
+}
+
+TEST(Tushentsova_Karina_Permut_Generator, CanCreateSerialArray) {
+    int const size = 5;
+    int a[size] = { 0 };
```

Comment:
```suggestion
    int a[size]{ 0 };
```
Также пожалуйста поправьте в других местах

Reply:
Все поправки внесены